ci/matrix.xml

```diff
@@ -232,6 +232,18 @@
     sbt ++$scala partest/compile
   ]]></task>
 
+  <task id="doc-generation"><![CDATA[
+    setJavaVersion $java
+    sbt ++$scala compiler/compile:doc library/compile:doc javalibEx/compile:doc \
+                 testInterface/compile:doc jasmineTestFramework/compile:doc stubs/compile:doc
+  ]]></task>
+
+  <task id="sbtplugin-doc-generation"><![CDATA[
+    setJavaVersion $java
+    sbt ++2.10.5 ir/compile:doc tools/compile:doc jsEnvs/compile:doc \
+                 testAdapter/compile:doc sbtPlugin/compile:doc
+  ]]></task>
+
```

Comment:
Ok. You got the idea :) Now this will fork off a totally new Jenkins task, which is probably overkill...

You can add doc generation for `ir`, `tools`, `jsEnvs`, `testAdapter` and `sbtPlugin` to the `tools-cli-stubs` / `tools-cli-stubs-sbtplugin` tasks and doc generation for `complier`, `library`, `javalibEx`, `testInterface`, `jasmineTestFramework` and `stubs` to `main` (you can also remove the line with `package` and `packageDocs` there, it is pointless).